internal/provider/resource_gitlab_personal_access_tokens_test.go

```diff
@@ -0,0 +1,109 @@
+package provider
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabPersonalAccessToken_basic(t *testing.T) {
+	testAccCheck(t)
+
+	user := testAccCreateUsers(t, 1)[0]
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabPersonalAccessTokenDestroy,
+		Steps: []resource.TestStep{
+			// Create a basic access token.
+			{
+				Config: fmt.Sprintf(`
+				resource "gitlab_personal_access_token" "foo" {
+					user_id = %d
+					name    = "foo"
+					scopes  = ["api"]
+				}
+				`, user.ID),
+				// Check computed and default attributes.
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("gitlab_personal_access_token.foo", "active", "true"),
+					resource.TestCheckResourceAttr("gitlab_personal_access_token.foo", "revoked", "false"),
+					resource.TestCheckResourceAttrSet("gitlab_personal_access_token.foo", "token"),
+					resource.TestCheckResourceAttrSet("gitlab_personal_access_token.foo", "created_at"),
+					resource.TestCheckResourceAttrSet("gitlab_personal_access_token.foo", "user_id"),
+					resource.TestCheckNoResourceAttr("gitlab_personal_access_token.foo", "expires_at"),
+				),
+			},
+			// Verify upstream resource with an import.
+			{
+				ResourceName:      "gitlab_personal_access_token.foo",
+				ImportState:       true,
+				ImportStateVerify: true,
+				// The token is only known during creating. We explicitly mention this limitation in the docs.
+				ImportStateVerifyIgnore: []string{"token"},
+			},
+			// Recreate the access token with updated attributes.
+			{
+				Config: fmt.Sprintf(`
+				resource "gitlab_personal_access_token" "foo" {
+					user_id    = %d
+					name       = "foo"
+					scopes     = ["api", "read_api", "read_repository", "write_repository"]
```

Comment:
Related to my other comment, can you make sure you include all valid scopes in this line?

Reply:
done also during this testing I found out something that is really off with the API. I will denote this in the thread above about scopes